grammar/grammar.py

```diff
@@ -222,6 +222,101 @@ def _get_productions_by_non_terminal(self, non_terminal):
         Returns a list of the productions from the passed non terminal.
         '''
         return [ p for n, p in self.productions if n == non_terminal ]
+    def has_direct_left_recursion(self):
+        non_terminal_with_left_recursion = set()
+        for prod in self.productions:
+            if prod.n == prod.p[0]:
+                non_terminal_with_left_recursion.add(prod.n)
+        return len(non_terminal_with_left_recursion) > 0, non_terminal_with_left_recursion
```

Comment:
Why this returns a tuple? If someone needs to 'test' the return value it can be only with the set returned.

Reply:
por default acho que se tu fizer if function ele pega o primeiro elemento da tupla então funcionaria normalmente, mas pros casos em que se quer saber os non terminais que tem recursao a esquerda fica mais facil

Reply:
Se tu fizer `if your_set` funciona também. Se o `set` for vazio, entende como falso.